agent/client/workflow_shutdown.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/kubeshop/tracetest/agent/proto"
+)
+
+func (c *Client) startShutdownListener() error {
+	ctx := context.Background()
+
+	client := proto.NewOrchestratorClient(c.conn)
+
+	request, err := c.getConnectionRequest()
+	if err != nil {
+		return err
+	}
+
+	stream, err := client.RegisterShutdownListener(ctx, request)
+	if err != nil {
+		return fmt.Errorf("could not open agent stream: %w", err)
+	}
+
+	go func() {
+		for {
+			resp := proto.ShutdownRequest{}
+			err := stream.RecvMsg(&resp)
+			if err == io.EOF {
+				return
+			}
+
+			if err != nil {
+				log.Fatal("could not get message from trigger stream: %w", err)
+			}
+
+			// TODO: get context from request
+			c.shutdownListener(context.Background(), &resp)
```

Comment:
Do you plan to do that in another PR?

Reply:
yes, several places require this. I want to have a PR to instrument the agent